test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/MediaTypeTest.cs

```diff
@@ -21,16 +23,30 @@ public void CanParse_ParameterlessMediaTypes(string mediaType)
             Assert.Equal(new StringSegment("json"), result.SubType);
         }
 
+        public static TheoryData<string> MediaTypesWithParameters
+        {
+            get
+            {
+                return new TheoryData<string>
+                {
+                    "application/json;format=pretty;charset=utf-8;q=0.8",
+                    "application/json;format=pretty;charset=\"utf-8\";q=0.8",
+                    "application/json;format=pretty;charset=utf-8; q=0.8 ",
+                    "application/json;format=pretty;charset=utf-8 ; q=0.8 ",
+                    "application/json;format=pretty; charset=utf-8 ; q=0.8 ",
+                    "application/json;format=pretty ; charset=utf-8 ; q=0.8 ",
+                    "application/json; format=pretty ; charset=utf-8 ; q=0.8 ",
+                    "application/json; format=pretty ; charset=utf-8 ; q=  0.8 ",
+                    "application/json; format=pretty ; charset=utf-8 ; q  =  0.8 ",
+                    " application /  json; format =  pretty ; charset = utf-8 ; q  =  0.8 ",
+                    " application /  json; format =  \"pretty\" ; charset = \"utf-8\" ; q  =  \"0.8\" ",
```

Comment:
Suspect your comment comes from the HTTP 1.1 specs. But the `Accept` header is much less constrained (barely mentioned) in [HTTP 1.0](https://www.w3.org/Protocols/HTTP/1.0/spec.html#Accept) and `q` could be used with that protocol (by convention).